docs/jwst/references_general/explain_rmaps.rst

```diff
@@ -0,0 +1,393 @@
+*reference_file_types* which defines the CRDS reference file types that are
+required for running the calibration step.  Note that for some Steps the
+reference file types actually used vary so the minimal list of required types
+may not be known if no science data is defined.
+
+CRDS Prefetch
+-------------
+
+To ensure all reference files required by a pipeline are available prior to
+processing, Pipelines perform a "pre-fetch" of all references required by any
+Step in the Pipeline.  This generic Pipeline behavior is intended to prevent
+processing which fails due to missing CRDS files after running for lengthy
+periods.
+
+When *CRDS_SERVER_URL* and *CRDS_PATH* are properly configured, CRDS will
+download and locally cache the minimal set of references required to calibrate
+specific data using a particular pipeline.  This configuration supports remote
+processing for users with no access or ineffecient access to the STScI local
+network.
+
+The pre-fetch also enables CRDS to report on all reference file assignment and
+availability problems a pipeline will encounter in a single CAL run.  This is
+required in I&T scenarios where the cadence of testing is roughly once per week
+and there is a need to
+
+While the prefetch runs for onsite users,  since the default CRDS configuration
+points to a complete CRDS cache,  no downloads will occurr.
+
+Step Method .get_reference_file()
+---------------------------------
+
+During processing individual Steps make secondary calls to CRDS via the
+*get_reference_file(input_file, reference_file_type)* method to fetch the cache
+paths of individual reference files.  If no file is applicable, CRDS returns
+the value 'N/A' which is sometimes used to skip related Step processing.  If
+there is an error determining a reference file, CRDS will raise an exception
+stopping the calibration; typically these occur due to missing reference files
+or incorrectly specified dataset parameters.
+
+While get_reference_file() returns the absolute path of CRDS reference files,
+reference file assignments are recorded in output products using a *crds://*
+URI prefix which translates to roughly "the path of this file in the local
+cache you've defined using CRDS_PATH,  or under /grp/crds/cache if you didn't
+define CRDS_PATH".
+
+Best Reference Matching
+=======================
+
+The Calibration Reference Data System (CRDS) assigns the best reference files
+needed to process a data set based on the dataset's metadata (FITS headers) and
+plain text CRDS rules files.
+
+CRDS rules and references are organized into a 4 tiered hierarchical network of
+versioned files consisting of:
+
+  * .pmap  - The overall context for the pipeline (i.e. all instruments)
+  * .imap  - The rules for all reference types of one instrument
+  * .rmap  - The rules for all references files of one type of one instrument
+  * .fits,.asdf,.json - Individual reference files assigned by .rmaps
```

Comment:
The indentation in the lines above makes a bar to appear at left as if it's a quoted text. 
It may look better if remove all spaces before `*`.  Unless you meant this.
To see what I mean click the `View` button above.

Reply:
Will fix.